File: 0268-missing-number/0268-missing-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    let set = new Set();
    
    for(let i = 0; i < nums.length; i++) {
        set.add(nums[i]);
    }
    
    for(let i = 0; i < nums.length + 1; i++) {        
        if(!set.has(i)) {
            return i;
        }
    }
    
    return -1;
};

File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        // sort the array
        Arrays.sort(nums);

        // loop over
        for(int i = 0; i < nums.length; i++) {
            int curr = nums[i];
            if(curr != i) return i;
        }

        // return ans if not found in loop
        return nums.length;
    }
}